inc/3rd-party/plugins/ecommerce/ithemes-exchange.php

```diff
@@ -0,0 +1,54 @@
+<?php
+defined( 'ABSPATH' ) or die( 'Cheatin\' uh?' );
+
+if ( function_exists( 'it_exchange_get_page_type' ) && function_exists( 'it_exchange_get_page_url' ) ) :
+	add_filter( 'rocket_cache_reject_uri', 'rocket_exclude_ithemes_exchange_pages' );
+	add_action( 'update_option_it-storage-exchange_settings_pages', 'rocket_after_update_single_options', 10, 2 );
+endif;
+
+function rocket_exclude_ithemes_exchange_pages( $urls ) {	
+	$pages = array(
+		'purchases',
+		'confirmation',
+		'account',
+		'profile',
+		'downloads',
+		'purchases',
+		'log-in',
+		'log-out'
+	);
+	
+	foreach( $pages as $page ) {
+		if ( it_exchange_get_page_type( $page ) == 'wordpress' ) {
+			$exchange_urls = get_rocket_i18n_translated_post_urls( it_exchange_get_page_wpid( $page ) );
+		} else {
+			$exchange_urls = array( parse_url( it_exchange_get_page_url( $page ), PHP_URL_PATH ) );
```

Comment:
Not mandatory, but someday it could be good to use `wp_parse_url()` everywhere. But beware:
1- The function was created in WP 4.4, so we need to add compatibility.
2- Its `$component` parameter was added in WP 4.7 ( -_- ), so we can't use it, otherwise it will explode for versions between 4.4 and 4.7.